controllers/metal3.io/host_state_machine.go

```diff
@@ -503,28 +503,18 @@ func (hsm *hostStateMachine) handleDeprovisioning(info *reconcileInfo) actionRes
 			hsm.Host.Status.ErrorCount = 0
 		}
 	} else {
-		skipToDelete := func() actionResult {
-			hsm.NextState = metal3v1alpha1.StateDeleting
-			info.postSaveCallbacks = append(info.postSaveCallbacks, deleteWithoutDeprov.Inc)
-			return actionComplete{}
-		}
-
 		switch r := actResult.(type) {
 		case actionComplete:
 			hsm.NextState = metal3v1alpha1.StateDeleting
 			hsm.Host.Status.ErrorCount = 0
-		case actionFailed:
-			// If the provisioner gives up deprovisioning and
-			// deletion has been requested, continue to delete.
-			if hsm.Host.Status.ErrorCount > 3 {
-				return skipToDelete()
```

Comment:
This feels like it might belong in a separate PR, so we can discuss (and if necessary some day revert) it independently.
The case where the underlying host just goes away is a fairly common one, and people justifiably complain about this making the BMH undeletable. So while I understand the argument, this is also an intentional feature we're removing.

Reply:
Deleting unmanaged hosts is impossible for other reasons, not because of cleaning IIRC. There is a bug report for that from people with cleaning disabled.

Splitting this change.